### PR TITLE
Remove nodes when displaying lines and shapes + make lines consistent across different map providers

### LINKED ISSUE #5565: Remove nodes when displaying lines and shapes
#### **User problem**

- The nodes make the shape and line look editable 
 
Screenshot 2023-04-28 at 4 05 56 PM

#### **Proposed Solution**
Design software or 3D programs remove the nodes to signify that it's not in edit mode, and I think we can apply the same concept here. 

- Remove nodes when displaying lines and shapes that aren’t editable. 
- We also want to make the lines and lines around the shape the same thickness (see in the image above)

Screenshot 2023-04-28 at 4 07 13 PM

#### History
We decided to keep the nodes before we added the selectable shading. Without node markers or shading, it was not very clear that the shape could be interacted with.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5565 

#### What has been done to verify that this works as intended?
I've tested the changes manually.

#### Why is this the best possible solution? Were any other approaches considered?
I've considered at least three other solutions here:
1. I thought about creating and displaying markers like we do now but making their icons transparent when needed. It would be probably the easiest solution to implement but not very elegant I would even call it a workaround, not a solution plus such a solution/workaround wouldn't bring any performance improvements. In my solution implemented in this pr I do not create markers we don't need so the app should work faster but with the one described here as I said markers would be still there but transparent.
2. Another solution was passing `draggable` (true/false) value to `polyline`/`polygon` features (like we do already in the case of `polyline` at least) and deciding whether we need to create lines and markers or lines only. It sounds like the best option and I even started implementing it but the code was full of if-else statements and it was difficult to read and work with.
3. The last solution was similar to the one implemented in this pr but assumed that different types of `polyline`/`polygon` (static/dynamic) create one hierarchy (for example `DynamicPolylineFeature` extends `StaticPolylineFeature` and maybe even `StaticPolylineFeature` extends abstract `PolylineFeature`. That would let us share some duplicated code that's true but again it would rather cause the code more complicated.

That's why I decided to have separate classes for every type of feature (dynamic/static).

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Please test carefully the geo widgets and select one from map with all three map providers of course.

#### Do we need any specific form for testing your changes? If so, please attach one.
I used the AllWidgetsForm for testing geo widgets and this one for selecting points/lines/shapes from map: [SelectOneFromMapForm.zip](https://github.com/getodk/collect/files/11665732/SelectOneFromMapForm.zip)

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: mapbox/src/main/java/org/odk/collect/mapbox/StaticPolygonFeature.kt
Comment (grzesiek2010) [inline mapbox/src/main/java/org/odk/collect/mapbox/StaticPolygonFeature.kt:25]: In mapbox it's not possible to set stroke width of a polygon. I would need to create additional polylines and add them there to make it look like one polygon (with borders). The only thing I was able to do was set this color so at least now the border is visible (it's thinner but it's there). I think it's ok for now. In the future, we can consider whether we want to fix it in mapbox too with the workaround I mentioned above or maybe remove the borders from all polygons at all.

**Review by seadowg — APPROVED** [review]

**Comment by dbemke** [comment]
I've noticed that it became more difficult to select a trace. On the master version it is easier even for the maximum zoom level. I don't know if it's about the zoom or because of removed nodes, thinner lines or maybe something Mapbox related but sometimes it takes tapping 10 times to select a trace on Mapbox and OSM. (Fake GPS used only for recording).

https://github.com/getodk/collect/assets/56479916/31c2d5a6-3131-4c11-b4ae-d882f2d6d7d5

**Comment by grzesiek2010** [comment]
We wanted to have the lines consistent across all three map engines so in Mapbox I've reduced the thickness. As a result, it might be more difficult to select such a line. We could slightly increase the thickness and then wait for feedback from users if it's enough. What do you think @seadowg?
But I think it might be in a separate pr so if you are done with testing @dbemke @srujner let's merge this pr.

**Comment by seadowg** [comment]
Agreed. Let's merge this once testing is finished and look at tweaking line thickness later (that'd be an easy hotfix option for example).

**Comment by dbemke** [comment]
I checked tapping the traces zoomed to streets level. In case of Mapbox selecting a trace takes: first try - tapping 3 times, second try- tapping 8 times, third try- tapping 6 times. I think users might feel that it's not working.

https://github.com/getodk/collect/assets/56479916/237fa3e3-97ad-4fde-8d39-c5f56ff2ee37

**Comment by seadowg** [comment]
Right, that's look pretty bad. Can we increase the thickness now @grzesiek2010? I don't think we need to be too concerned about the engines being consistent.

**Comment by grzesiek2010** [comment]
Ok, I'm playing with improving it.

**Comment by grzesiek2010** [comment]
Ok I made the lines thicker. I hope it works better now but keep in mind that when it comes to lines it will never be perfect so we have to find a good balance.

**Comment by srujner** [comment]
**Tested with Success!**

**Verified on device with Android:** 12, 13

**Verified cases:**

- Regression checks for GeoPoint, GeoTrace, GeoShape widgets;
- Nodes has been removed;
- Lines are thicker :) and easier to select;
- Select Map form;
- Offline Layers;
- Google Maps, Mapbox, OSM;
- Collecting Location in background;
- View Submissions on map;
- Zooming in and out to location
- Screen rotation, minimize app, using backward button;
- Light and Dark mode.

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android: 10

**Pull request merged**
